components/gateway/internal/auditlog/worker.go

```diff
@@ -33,6 +35,7 @@ func (w *Worker) Start() {
 			return
 		case msg := <-w.auditlogChannel:
 			log.Printf("Read from auditlog channel (size=%d, cap=%d)", len(w.auditlogChannel), cap(w.auditlogChannel))
+			w.collector.SetChannelSize(len(w.auditlogChannel))
```

Comment:
Can there be a race condition here? Or those Prometheus objects (prometheus.Gauge in our case) are synchronized?

Reply:
prometheus Gauge is safe for concurrent use as it uses atomic operations